### DISCUSSION TITLE
Is not caching action reqs a good idea?

### DISCUSSION THREAD

**Discussion by timkelty** [ostark/upper#49] [General]
By default, we don't cache action reqs: https://github.com/ostark/upper/blob/bd6cd34ffc96e8cb8971316d0f680157a6762417/src/EventRegistrar.php#L65

For example, we're using Sprig on a site, and we want the responses cached. While the responses are all GETs with query params, they all have `Cache-Control: private`, because they are action reqs.

I'm wondering why, and if this is  necessary at all?

Seems like any action reqs that _weren't_ cacheable would already be naturally, either by being a POST or having a `set-cookie` header triggered from a `session_start()`;

**Comment by timkelty** [comment]
This was added here: https://github.com/ostark/upper/issues/36

**Comment by timkelty** [comment]
https://github.com/ostark/upper/issues/36#issuecomment-583071931 - I think this reasoning was flawed, in that those will initiate sessions and have a set-cookie in the response, and therefore should be already be un-cacheable.

Perhaps a safer change would be to check if the response has a Set-Cookie, and if so, set `Cache-control: private`?

**Reply by ostark** [reply]
hmm, I'll consider this!
